torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -2290,12 +2313,18 @@ def forward(self, *args: Any, **kwargs: Any) -> Any:
                     input_dtype, *args, **kwargs
                 )
 
-            # All-gather full parameters, moving them to compute_device if
-            # necessary.
-            self._rebuild_full_params()
+            # Only rebuilding full params when the params are not prefetched in previous layers
+            if not self._forward_full_params_prefetched:
+                self._rebuild_full_params()
```

Comment:
we can always call 'self._rebuild_full_params()' here, before calling it, check if _need_prefetch_full_params(), wait for all gather stream, so that "_rebuild_full_params" will not trigger a duplicated all gather

Reply:
Done